Sources/Macros/MacroDefinitions.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  MacroDefinitions.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Compile-time validated URL instantiation
```

Comment:
@mallexxx I think we should work on the documentation of #URL macro a little bit before it gets released. From the user perspective I've found a few things that weren't obvious:
1. Classic `http://💩.la` is not valid
2. File URLs seem to be out of scope of this macro (but a URL with `file://` scheme works)

Could you please update the docs to provide more information about what to expect from the macro, when to not use it, and any further information about it that you consider important?

Should we consider adding a swiftlint rule to warn about `URL.init(string:)` initializer use?

Reply:
the idea is to disable any applied URL conversions like punycode or no-scheme urls, only 1-1 mapped valid URLs, I‘ll add some docs

Of course I considered adding some swiftlint rule but it seems more of a task for a static analyzer as trying to apply a regular regex rule hogs the build time too much

Reply:
> the idea is to disable any applied URL conversions like punycode or no-scheme urls, only 1-1 mapped valid URLs

This already sounds great (as part of the documentation) - makes it clear when the macro can be used. So the bottom line is, we still consider `URL.init(string:)` to still have valid use cases, such as when translating user input to URL, etc. In this case I don't think we even need to think about the static analysis. Thanks for clarifying it!